jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/LogCleanupCommand.java

```diff
@@ -268,11 +265,40 @@ public ExecutionResults execute(CommandContext ctx) throws Exception {
                 executionResults.setData("ProcessInstanceLogRemoved", piLogsRemoved);
             }
         }
+        calculateNextRun(ctx);
         executionResults.setData("BAMLogRemoved", 0L);
         return executionResults;
     }
     
     private  static boolean mightBeMore (int deleted, int recordPerTransaction) {
         return recordPerTransaction > 0 && deleted >= recordPerTransaction;
     }
+
+    private void calculateNextRun(CommandContext ctx) {
+        String singleRun = (String) ctx.getData("SingleRun");
+        if ("true".equalsIgnoreCase(singleRun)) {
+            // disable rescheduling
+            nextScheduleTimeAdd = -1;
+            return;
+        }
+
+        String nextRun = (String) ctx.getData("NextRun");
+        if (nextRun != null) {
+            nextScheduleTimeAdd = DateTimeUtils.parseDateAsDuration(nextRun) - calculateExecutionTimeInMillis(ctx);
```

Comment:
@gmunozfe thanks for the input, that makes sense. I added the check that in case there is an overlap (previous execution took longer than nextRun value), the next execution is scheduled immediately. Also added a new test case for this scenario.